distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/AdminConfig.java

```diff
@@ -74,4 +82,33 @@ public void setLogLevels() throws IOException {
 
         logConfig.update(properties);
     }
+
+    public void updateConfig(ConfigWaitable configWaitable)
+            throws IOException, InterruptedException {
+        getDdfConfigAdmin().update(configWaitable.getPid(), configWaitable.getConfigProps());
+
+        long timeoutLimit = System.currentTimeMillis() + CONFIG_UPDATE_MAX_WAIT_MILLIS;
+        while (true) {
+            if (configWaitable.isConfigured()) {
+                break;
+            } else {
+                if (System.currentTimeMillis() > timeoutLimit) {
+                    fail(String.format("Timed out waiting for configuration change for %s",
+                            configWaitable.getPid()));
+                } else {
+                    Thread.sleep(LOOP_SLEEP_MILLIS);
+                }
+            }
+        }
+    }
+
+    public interface ConfigWaitable {
```

Comment:
I wonder if this class should be extracted and renamed SynchronizedConfiguration.  You can then move the updateConfig method to the class.  You could also provide a default implementation that takes a Callable<Boolean> in the constructor for the isConfigured().  Thoughts?